Integrate nrfx 3.6.0 release

nika-nordic opened the pull request:
Integrate new nrfx 3.6.0 release that adds support for nRF54L20 Eng A and nRF9230 Eng B devices.

nika-nordic committed:
manifest: update hal_nordic revision to integrate nrfx 3.6.0

New nrfx version contains support
for nRF54L20 Eng A and nRF9230 Eng B devices.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic committed:
modules: hal_nordic: nrfx: remove nrfx_uarte rx cache for nrf54l15 enga

This option is no longer valid for nRF54L15 Eng A target.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic committed:
drivers: serial: nrf: add default value for frame timeout cfg

This is needed to avoid warnings about uninitialized
structure member, which was added in nrfx 3.6.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

anangl commented on modules/hal_nordic/nrfx/nrfx_config_nrf54l15_enga_application.h:
```diff
@@ -1517,15 +1517,6 @@
 #endif
 
-/**
- * @brief NRFX_UARTE_CONFIG_RX_CACHE_ENABLED
```
This should be also removed: https://github.com/zephyrproject-rtos/zephyr/blob/ef4747c939231bab5603c66cec5054219dbc6513/modules/hal_nordic/nrfx/nrfx_config.h#L815-L817
and this: https://github.com/zephyrproject-rtos/zephyr/blob/ef4747c939231bab5603c66cec5054219dbc6513/modules/hal_nordic/nrfx/Kconfig#L951-L957

nika-nordic replied:
IMO it shouldn't be removed, because nrf54h_* devices still make use of it - it was only removed for nrf54l_* devices. Please take a look at diff for nrfx_config template for nrf54h app, for example: https://github.com/zephyrproject-rtos/hal_nordic/commit/884c4d61746bc35fbd379c169fc87ddb56c6461d#diff-02800d39e6b44178911a9be5fda6ec7bdf90fd37e76a1f8687ab39f433653659

anangl replied:
Ok, I somehow thought that this feature was removed entirely from the driver. Now I see that I misunderstood.

anangl reviewed (commented):
`nrfx_config*.h` files should also be updated with addition of the `NRFX_UARTE_RX_FIFO_FLUSH_WORKAROUND_MAGIC_BYTE` option, to follow changes done to the template files in the hal_nordic module.

nika-nordic committed:
modules: hal_nordic: nrfx: sync configs for UARTE MAGIC_BYTE option

New option is used in RX FIFO flush workaround
performed by the nrfx_uarte driver.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic commented:
@anangl

> nrfx_config*.h files should also be updated with addition of the NRFX_UARTE_RX_FIFO_FLUSH_WORKAROUND_MAGIC_BYTE option, to follow changes done to the template files in the hal_nordic module.

done
